src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -149,14 +152,32 @@ public void deleteCollection(String collectionName) throws PineconeException {
         }
     }
 
-    public Index createIndexConnection(String indexName) {
-        PineconeConnection connection = new PineconeConnection(config, indexName);
-        return new Index(connection);
+    public Index getIndexConnection(String indexName) {
+        config.setHost(getIndexHost(indexName));
+        PineconeConnection connection = getConnection(indexName);
+        return new Index(this, connection, indexName);
+    }
+
+    public AsyncIndex getAsyncIndexConnection(String indexName) {
+        config.setHost(getIndexHost(indexName));
+        PineconeConnection connection = getConnection(indexName);
+        return new AsyncIndex(this, connection, indexName);
+    }
```

Comment:
For example, we should validate `indexName` is non-empty here.

Reply:
Right, there's no need to attempt creating a connection